src/metrics/metrics.ts

```diff
@@ -492,6 +484,44 @@ export async function getPieceLevelMoveInfo(games: FileReaderGame[]) {
     piecesWithMostMovesInAGame,
     gameLinksWithPiecesMostMoves,
     numMovesMadePieceWithMostMoves,
+  };
+}
+
+export async function getPiecePromotionInfo(games: FileReaderGame[]) {
```

Comment:
I think we should just start thinking ahead now so we don't have to refactor later. Rather than take a list of games most functions should just maintain a history and take in a single game, but be called repeatedly.

We could rewrite these as a class to do this, and have the class store member variables or we could pass maps/objects that could potentially be shared by the functions.

I'm probably a fan of the latter

Reply:
why not allow it to take a list of games? why is this a necessary refactoring?

see my message in discord re: conducting the large analysis. I think if option 1, taking in a list of games is ok https://discord.com/channels/1156826703440646145/1169163573759578123/1169501929915822110

Reply:
are you saying you want this:

- filereader function to parse the pgn, spit out history for each game to metrics functions
- metrics functions take in history for single game, conduct analysis
- index function to repeatedly call metrics functions and run comprehensive analysis?

Reply:
best will be to write an example, i'll do it somewhere besides here